### ISSUE TITLE
feat: `TypeAliasNode.getProperties()` like `InterfaceNode.getProperties()`

### ISSUE THREAD

**Issue by mikestopcontinues** [jordimarimon/ts-ast-parser#213]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/jordimarimon/ts-ast-parser/blob/main/CONTRIBUTING.md).
- [X] I have searched for [existing issues](https://github.com/jordimarimon/ts-ast-parser/issues) that already report this problem, without success.

### Describe the Feature Request

`InterfaceNode` provides a nice way to access the properties defined on it, but `TypeAliasNode` does not. However, when `TypeAliasNode` is an `ObjectLiteral`, it has a `Field[]` like `InterfaceNode`.

### Describe the Use Case

Currently, in order to handle both declaration types, I have to `.serialize()`, but since the two forms are often used interchangeably in TS, it would be great to have a parallel method of handling both.

**Comment by jordimarimon** [comment]
When you say a "parallel method of handling both" what are you thinking of? Is to have a better idea so I can evaluate if it's something feasible to have.

These are the two files involved:

- [InterfaceNode](https://github.com/jordimarimon/ts-ast-parser/blob/main/packages/core/src/nodes/interface-node.ts)
- [TypeLiteral/ObjectLiteral](https://github.com/jordimarimon/ts-ast-parser/blob/main/packages/core/src/types/type-literal-node.ts)

Both have a `getProperties()` method but obviously you will have to first narrow `ReflectedTypeNode` to `TypeLiteralNode` to be able to use the `getProperties()` method because not all types are object literals. I can see why this may not be something nice but right now I don't see what could be done to simplified it.

**Comment by mikestopcontinues** [comment]
Thanks! You helped me to see the problem. I think there's any issue with the parameter type of `is.TypeLiteralNode()`. Here's the issue I was working around:

```ts
  let p: PropertyNode[] = [];

  if (node && is.InterfaceNode(node)) {
    p = node.getProperties();
  } else if (node && is.TypeAliasNode(node)) {
    const type = node.getValue();

    // TYPE ERROR
    // Argument of type 'ReflectedNode<Type, Node>' 
    // is not assignable to parameter of type 'ReflectedTypeNode<TypeNode>'.
    if (is.TypeLiteralNode(type)) {
      p = type.getProperties();
    }
  }
```

However, `if (type instanceof TypeLiteralNode)` properly validates that the `type` is what I expect it to be. And that unlocks `.getProperties()`. Perhaps this would be solved by #212?

**Comment by mikestopcontinues** [comment]
Many, many thanks!
